### PR TITLE
 Add searchable snapshots integration tests for URL repositories

### PR DESCRIPTION
**Pull request by fcofdez**
This commit also increases the max number of pooled HTTP connections

### CODE REVIEW

File: test/fixtures/url-fixture/src/main/java/fixture/url/URLFixture.java [hunk 1]
```diff
@@ -18,13 +18,15 @@
...
 import java.util.HashMap;
 import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * This {@link URLFixture} exposes a filesystem directory over HTTP. It is used in repository-url
  * integration tests to expose a directory created by a regular FS repository.
  */
 public class URLFixture extends AbstractHttpFixture {
```
Comment (ywelsch) [inline test/fixtures/url-fixture/src/main/java/fixture/url/URLFixture.java:28]: This is a fixture. I was wondering if we could plug in Nginx or some other simple http server to really make sure it works with at least a proper http server.

**Review by ywelsch — APPROVED** [review]
Thanks for this PR! I've left one suggestion which I think will give us more confidence that the client is behaving well.

File: test/fixtures/url-fixture/src/main/java/fixture/url/URLFixture.java (hunk 1 above)
Comment (fcofdez) [reply test/fixtures/url-fixture/src/main/java/fixture/url/URLFixture.java:28]: That's a good idea, I'll add a real http server to the test suite 👍

**Review by ywelsch — COMMENTED** [review]
I've left two build-related comments. I think someone from @elastic/es-delivery should have a look here too.

File: x-pack/plugin/searchable-snapshots/qa/url/build.gradle [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  setting 'path.repo', repositoryDir.absolutePath
+  setting 'repositories.url.allowed_urls', { "${-> fixtureAddress('nginx-fixture')}" }, IGNORE_VALUE
+
+  setting 'xpack.license.self_generated.type', 'trial'
+
+  setting 'xpack.searchable.snapshot.shared_cache.size', '16MB'
+  setting 'xpack.searchable.snapshot.shared_cache.region_size', '256KB'
+}
+
+tasks.register("urlThirdPartyTest") {
```
[the remark this answers is not in the document]
Comment (fcofdez) [reply x-pack/plugin/searchable-snapshots/qa/url/build.gradle:49]: Yes, that was a mistake.

**Comment by fcofdez** [comment]
@elasticmachine run elasticsearch-ci/eql-correctness
There was a network timeout unrelated to the changes

File: test/fixtures/nginx-fixture/build.gradle [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+// These directories are shared between the URL repository and the FS repository in integration tests
+project.ext {
+  fsRepositoryDirs = [
+    'nginx-fixture': new File(project.buildDir, "shared-repository"),
+  ]
+}
+
+for (repoDir in project.ext.fsRepositoryDirs) {
+  delete(repoDir.value as File)
```
[the remark this answers is not in the document]
Comment (mark-vieira) [reply test/fixtures/nginx-fixture/build.gradle:20]: We can delete the directory contents in `preprocessfixture` task if necessary. We definitely don't want to do this during configuration time.

**Comment by fcofdez** [comment]
@elasticmachine update branch

**Review by ywelsch — APPROVED** [review]
LGTM. Thanks for adding these tests!

**Pull request merged**